EpiAware/test/predictive_checking/fast_approx_for_r.jl

```diff
@@ -0,0 +1,73 @@
+#=
+# Fast approximation for `r` from `R₀`
+I use the negative moment generating function (MGF).
+
+Let
+```math
+G(r) = \sum_{i=1}^{\infty} w_i e^{-r i}.
+```
+and
+```math
+f(r, \mathcal{R}_0) = \mathcal{R}_0 G(r) - 1.
+```
+then the connection between `R₀` and `r` is given by
+```math
+f(r, \mathcal{R}_0) = 0.
+```
+Given an estimate of $\mathcal{R}_0$ we implicit solve for $r$ using a root
+finder algorithm. In this note, I test a fast approximation for $r$ which
+should have good autodifferentiation properties. The idea is to start from the
+small $r$ approximation to the solution of $f(r, \mathcal{R}_0) = 0$ and then
+apply one step of Newton's method. The small $r$ approximation is given by
+```math
+r_0 = { \mathcal{R}_0 - 1 \over  \mathcal{R}_0 \langle W \rangle }.
+```
+where $\langle W \rangle$ is the mean of the generation interval.
+
+To rapidly improve the estimate for `r` we use Newton steps given by
+```math
+r_{n+1} = r_n - {\mathcal{R}_0 G(r) - 1\over \mathcal{R}_0 G'(r)}.
+```
+
+=#
+
+using TestEnv
+TestEnv.activate()
+using EpiAware
+using Distributions
+using StatsPlots
+
+# Create a discrete probability mass function (PMF) for a negative binomial distribution
+# with left truncation at 1.
+
+w =
+    create_discrete_pmf(NegativeBinomial(2, 0.5), D = 20.0) |>
```

Comment:
is it good style to start this on a new line?

Reply:
Fixed